Develop leaderboard page

mariarykova opened the pull request.

Commit:
Add stylus_testnet to the networks

add logo, title, description

add banner, table, footer

table style, banner style, background style

add styles to leader board, graphql requests

add fonts, change scss classes, table compontent

add wigme provider

add data to table from graph, add sorting, links

update sort function, update data receiving

add ens name, fix animation for social

add new queries, update filtering, sorting, receiving ensNames

add table styles, update queries, add context

add connectWalletModal, ethereumProvider

update wallet lists to connect

add data to frozen row, correct styles

add styles to mobile version, add some cypress tests

add more cypress tests

make some style changes, add docker file and makefile

changes according to code review comments

changes according to code review comments, add cypress tests

update tests, install Sentry

Start to support the leaderboard with a deployment

delete duplicated text on main page

add sentry variables, fix favicon

Start to support the leaderboard with a deployment

delete duplicated text on main page

add sentry variables, fix favicon

eli-d commented on web/leaderboard/public/assets/fonts/aeonik/Aeonik-Bold.woff:
remove all fonts from the repo, there should be a static import (check how app.fluidity.money does this)

eli-d commented on web/leaderboard/README.md:
```diff
@@ -0,0 +1,36 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
write a small description of what this does, how to build it, how to use it - remove the boilerplate

eli-d commented on web/leaderboard/config.toml:
```diff
@@ -0,0 +1,418 @@
+# Setup Webapp Network Drivers
+[[drivers.ethereum]]
```
delete this since it's not used

eli-d commented on web/leaderboard/cypress/e2e/page.cy.ts:
```diff
@@ -0,0 +1,33 @@
+      expect(response.status).to.eq(200);
+      expect(response).to.have.property("headers");
+      expect(response).to.have.property("duration");
+      expect(response.body).to.have.length(3);
```
is this always true?

eli-d commented on web/leaderboard/cypress/e2e/spec.cy.ts:
```diff
@@ -0,0 +1,5 @@
+describe('template spec', () => {
+  it('passes', () => {
+    cy.visit('https://example.cypress.io')
```
example test

eli-d commented on web/leaderboard/cypress/fixtures/example.json:
```diff
@@ -0,0 +1,5 @@
+{
```
example test

eli-d commented on web/leaderboard/cypress/support/commands.ts:
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
delete this file if it's not used

mariarykova replied:
I left this file but as it is needed for cypress tests running otherwise there was a mistake

eli-d commented on web/leaderboard/ethereum.d.ts:
```diff
@@ -0,0 +1,3 @@
+interface Window {
+  ethereum?: any; // Use `any` or a more specific type if available
```
shouldn't need to use `window.ethereum`

eli-d commented on web/leaderboard/package.json:
```diff
@@ -0,0 +1,39 @@
+    "next-urql": "^5.0.2",
+    "react": "^18",
+    "react-dom": "^18",
+    "urql": "^4.0.6",
```
remove this as a dependency

eli-d commented on web/leaderboard/public/assets/wallets/coin98.svg:
we should have all the logos we need in static hosting

eli-d commented on web/leaderboard/public/next.svg:
shouldn't need this

eli-d commented on web/leaderboard/public/vercel.svg:
don't need this

eli-d commented on web/leaderboard/src/app/components/Card/Card.module.scss:
fix surfing imports rather than duplicating these

mariarykova replied:
can't fix this, still have a problem

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectWalletModal/ConnectWalletModal.tsx:
```diff
@@ -0,0 +1,161 @@
+const ConnectWalletModal = ({ visible, close }: IConnectWalletModal) => {
+  const [modal, setModal] = useState<React.ReactPortal | null>(null);
+
+  const [configData, setConfigData] = useState<any>(null);
```
don't use any type

eli-d replied:
don't think this is used, delete it

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectWalletModal/ConnectWalletModal.tsx:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    fetchConfig();
+  }, []);
```
delete this since config isn't used

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectWalletModal/ConnectWalletModal.tsx:
```diff
@@ -0,0 +1,161 @@
+            </div>
+
+            <ul className={styles["connect-wallet-modal-list"]}>
+              {configData && <EthWalletsMap />}
```
remove configData here

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWallet/ConnectedWallet.tsx:
```diff
@@ -0,0 +1,57 @@
+import Jazzicon, { jsNumberForAddress } from "react-jazzicon";
+import styles from "./ConnectedWallet.module.scss";
+
+type Address = string;
```
unnecessary since it's only in this file

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWallet/ConnectedWallet.tsx:
```diff
@@ -0,0 +1,57 @@
+import styles from "./ConnectedWallet.module.scss";
+
+type Address = string;
+const addressSize = 12 / 2;
```
should just be set to 6, and have a comment as to why

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWallet/ConnectedWallet.tsx:
```diff
@@ -0,0 +1,57 @@
+
+// trimAddress formats Address to abcdef...
+const trimAddressShort = (address: Address): Address => {
+  const leftSide = address.substring(0, 5);
```
should use the constant from above?

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWallet/ConnectedWallet.tsx:
```diff
@@ -0,0 +1,57 @@
+    >
+      <div className={styles.holo}>
+        <img src="/user.svg" alt="user" />
+        {/*<Jazzicon diameter={36} seed={jsNumberForAddress(address)} />*/}
```
remove

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWalletModal/ConnectedWalletModal.tsx:
```diff
@@ -0,0 +1,143 @@
+  disconnect: () => void;
+}
+
+export const ConnectedWalletModal = ({
```
have this in the same file as connect wallet, and conditionally render

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWalletModal/ConnectedWalletModal.tsx:
```diff
@@ -0,0 +1,143 @@
+}: IPropsConnectedWalletModal) => {
+  const [modal, setModal] = useState<React.ReactPortal | null>(null);
+  const [icon, setIcon] = useState<React.ReactNode>(
+    <img height="32" width="32" src="/copyIconCircle.svg" alt="copy" />
```
have this state be something like `useState<"circle" | "checked" | ...>`, then check the condition when rendering

eli-d commented on web/leaderboard/src/app/components/ConnectWallet/ConnectedWalletModal/connectedWalletModal.cy.tsx:
```diff
@@ -0,0 +1,78 @@
+      disconnect: cy.stub().as("disconnect"),
+    };
+    cy.mount(<ConnectedWalletModal {...props} />);
+    cy.contains(trimAddress(props.address)).should("be.visible");
```
i would hardcode the trimmed address in the test file, since this doesn't really test that it trims correctly

eli-d commented on web/leaderboard/src/app/components/Profile/index.tsx:
```diff
@@ -0,0 +1,70 @@
+        <ConnectedWallet
+          address={String(address)}
+          callback={() => {
+            !connectedWalletModalVisibility &&
```
this is redundant

eli-d commented on web/leaderboard/src/app/components/Profile/index.tsx:
```diff
@@ -0,0 +1,70 @@
+          <ConnectedWalletModal
+            visible={connectedWalletModalVisibility}
+            close={() => setConnectedWalletModalVisibility(false)}
+            address={`0x${address}`}
```
don't think this needs a 0x in front

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+import styles from "../../page.module.scss";
+
+type Filter<T> = {
+  filter: (item: T) => boolean; // eslint-disable-line no-unused-vars
```
this shouldn't need an eslint tag

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+
+  const { rowsPerPage, page, paginate } = pagination;
+
+  //  const isTransition = useTransition();
```
remove

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+
+  const setSearchParams = useSearchParams();
+
+  const cappedPageCount = Math.min(240, count);
```
comment why 240

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+
+  const cappedPageCount = Math.min(240, count);
+
+  const pageCount = Math.ceil(cappedPageCount / rowsPerPage);
+
+  const startIndex = (page - 1) * rowsPerPage + 1;
+  const endIndex = Math.min(page * rowsPerPage, cappedPageCount);
+
+  const frozenRows = data.filter((row) => freezeRow?.(row));
```
these should all be memoised (`useMemo`)

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+    ({ show }) => show === undefined || show
+  );
+
+  const handlePageTurn = (page: number) => {
```
this function doesn't do anything

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+            <motion.tbody
+              key={`page-${page}`}
+              //  initial="enter"
+              // TODO: need to think what to use instead of useTransition from remix
```
address this TODO

eli-d commented on web/leaderboard/src/app/components/Table/index.tsx:
```diff
@@ -0,0 +1,300 @@
+              ))}
+              {/* Unfrozen Rows */}
+              {data
+                .filter((_) => !freezeRow?.(_))
```
don't name this `_` since that's usually used for unused variables, but here it's being used

eli-d commented on web/leaderboard/src/app/config.ts:
```diff
@@ -0,0 +1,60 @@
+import localFont from "next/font/local";
+
+export const aeonik = localFont({
```
use the static fonts rather than local

eli-d commented on web/leaderboard/src/app/config.ts:
```diff
@@ -0,0 +1,60 @@
+  ],
+});
+
+export const tableHeadings = [
```
these should be in Table

eli-d commented on web/leaderboard/src/app/context.tsx:
```diff
@@ -0,0 +1,120 @@
+import { utils, BigNumber, constants } from "ethers";
+
+import type BN from "bn.js";
```
take this out of surfing

eli-d commented on web/leaderboard/src/app/ensName.tsx:
```diff
@@ -0,0 +1,25 @@
+  return leftSide + "..." + rightSide;
+};
+
+const UseEnsName = (address: any) => {
```
don't use the any type

eli-d commented on web/leaderboard/src/app/ensName.tsx:
```diff
@@ -0,0 +1,25 @@
+
+const UseEnsName = (address: any) => {
+  const ensName = useEnsName({
+    address: address.address,
```
address should be passed directly to this hook

eli-d commented on web/leaderboard/src/app/ensName.tsx:
```diff
@@ -0,0 +1,25 @@
+const UseEnsName = (address: any) => {
+  const ensName = useEnsName({
+    address: address.address,
+    chainId: 1,
```
don't hardcode the chain (pass it in with a default maybe)

eli-d commented on web/leaderboard/src/app/favicon.ico:
should be static

eli-d commented on web/leaderboard/src/app/layout.tsx:
```diff
@@ -0,0 +1,24 @@
+  return (
+    <html
+      lang="en"
+      className={`${inter.className} ${spline.className} ${aeonik.className}`}
```
remove fonts

eli-d commented on web/leaderboard/src/pages/api/users.ts:
```diff
@@ -0,0 +1,40 @@
+  yield_earned: number;
+};
+
+export default function users(
```
remove test file

eli-d commented on web/leaderboard/src/pages/api/config.ts:
```diff
@@ -0,0 +1,55 @@
+import { parse } from "toml";
```
don't need this file

eli-d commented on web/leaderboard/src/config/webapp.config.server.ts:
```diff
@@ -0,0 +1,51 @@
+import { parse } from "toml";
```
don't need this file

eli-d commented on web/leaderboard/src/app/queries/useLeaderboardRanking.ts:
```diff
@@ -0,0 +1,185 @@
+  query LeaderboardRanking {
+    leaderboard_ranking(
+      args: { i: "1 day", network_: "arbitrum" }
+      limit: 40
```
pass limit in as a variable

eli-d commented on web/leaderboard/src/app/queries/useLeaderboardRanking.ts:
```diff
@@ -0,0 +1,185 @@
+  query LeaderboardRanking {
+    leaderboard_ranking(
+      args: { network_: "arbitrum" }
+      limit: 40
```
pass limit in as a variable

eli-d commented on web/leaderboard/src/app/queries/useLeaderboardRanking.ts:
```diff
@@ -0,0 +1,185 @@
+  query LeaderboardRanking {
+    leaderboard_ranking(
+      args: { i: "1 day", network_: $network }
+      limit: 40
```
pass limit in as a variable

eli-d commented on web/leaderboard/src/app/queries/ChainContext.tsx:
```diff
@@ -0,0 +1,118 @@
+export type Network = "arbitrum" | "etheruim" | "solana";
+
+interface ApiState {
+  leaderboardRanking24Hours: LeaderboardRanking[] | [];
```
don't need `| []`

eli-d commented on web/leaderboard/src/app/queries/ChainContext.tsx:
```diff
@@ -0,0 +1,118 @@
+import { useQuery } from "@apollo/client";
+import { useAccount, useConnect, useDisconnect, useEnsAvatar } from "wagmi";
+
+export type Network = "arbitrum" | "etheruim" | "solana";
```
ethereum misspelled, also this should exist somewhere else (probably in surfing)

eli-d commented on web/leaderboard/src/app/queries/ChainContext.tsx:
```diff
@@ -0,0 +1,118 @@
+
+interface ChainState {
+  network: Network;
+  userAddress: string | undefined;
```
prefer `userAddress?: string`

eli-d commented on web/leaderboard/src/app/queries/ChainContext.tsx:
```diff
@@ -0,0 +1,118 @@
+
+  const { address: userAddress, connector, isConnected } = useAccount();
+
+  const [onChainData, setOnChainData] = useState<any>({
```
don't use any type

eli-d commented on web/leaderboard/src/app/providers.tsx:
```diff
@@ -0,0 +1,63 @@
+    new WalletConnectConnector({
+      chains,
+      options: {
+        projectId: "60cd988a2683be6b1787c53081016310",
```
make into an environment variable

eli-d commented on web/leaderboard/src/app/page.tsx:
```diff
@@ -0,0 +1,446 @@
+import UseEnsName from "./ensName";
+import { Profile } from "./components/Profile";
+
+export type IRow = React.HTMLAttributes<HTMLDivElement> & {
```
is this duplicated?

eli-d commented on web/leaderboard/src/app/page.tsx:
```diff
@@ -0,0 +1,446 @@
+  });
+
+  const [sortedData, setSortedData] = useState<Data[]>([]);
+  const [sortedByItem, setSortedByItem] = useState<string>("number");
```
maybe make this an enum

eli-d commented on web/leaderboard/src/app/page.tsx:
```diff
@@ -0,0 +1,446 @@
+      })
+    );
+    if (isConnected) {
+      if (apiState.dataUser24Hours.length === 0) {
+        apiState.leaderboardRanking24Hours &&
+          apiState.dataUser24Hours &&
+          setSortedData(() => [...leaderboard, user]);
+      } else {
+        const newArray = [...leaderboard, ...apiState.dataUser24Hours];
+        apiState && setSortedData(newArray);
+      }
+    } else {
+      loading24Hours === false &&
+        setSortedData(apiState.leaderboardRanking24Hours);
```
this could be moved to a closure so it's not repeated below

eli-d commented on web/leaderboard/src/app/page.tsx:
```diff
@@ -0,0 +1,446 @@
+          setSortedData(() => [...leaderboard, user]);
+      } else {
+        const newArray = [...leaderboard, ...apiState.dataUser24Hours];
+        apiState && setSortedData(newArray);
```
this could be one line `setSortedData([...leaderboard, ...apiState.dataUser24Hours])`

eli-d commented on web/leaderboard/src/app/page.tsx:
```diff
@@ -0,0 +1,446 @@
+          const newSortedData = sortedData.sort(
+            (a, b) => Number(b.volume) - Number(a.volume)
+          );
+          //  const leaderboard = newSortedData.map((leaderboardRow, i) => ({
```
remove

eli-d commented on web/leaderboard/src/app/page.tsx:
```diff
@@ -0,0 +1,446 @@
+};
+
+export default function Home() {
+  const [filterIndex, setFilterIndex] = useState(0);
```
make this more clear (e.g. `useState<"allTime" | "24hour">` or something)

af-afk commented on web/leaderboard/sentry.client.config.ts:
```diff
@@ -0,0 +1,30 @@
+import * as Sentry from "@sentry/nextjs";
+
+Sentry.init({
+  dsn: "https://02225ae4a93c4b438998f8f56041566b@o1103433.ingest.sentry.io/6130103",
```
Set this with an environment variable!

af-afk commented on web/leaderboard/sentry.edge.config.ts:
```diff
@@ -0,0 +1,16 @@
+import * as Sentry from "@sentry/nextjs";
+
+Sentry.init({
+  dsn: "https://02225ae4a93c4b438998f8f56041566b@o1103433.ingest.sentry.io/6130103",
```
Environment variable here too!

mariarykova committed:
delete package-lock.json to resolve conflicts

mariarykova committed:
update styles, tried to fix npm-test

Commit:
Use the latest version of Node

Use node20 with v4 configure aws credentials

-Remove node20 from the credentials name

Commit:
Use node 20 instead of 16 in the setup node functino

mariarykova committed:
Add testEnvironment to solve npm-test error

mariarykova committed:
Merge branch 'develop-leaderboard-project' of github.com:fluidity-money/fluidity-app into develop-leaderboard-project

mariarykova committed:
delete testEnvironment

mariarykova committed:
remove installGlobals to watch how npm-test run

mariarykova committed:
delete tests warnings

mariarykova committed:
return installGlobals into tests

mariarykova committed:
add urql@4 dependancy
